src/FluentCheck.ts

```diff
@@ -162,24 +167,18 @@ abstract class FluentCheckQuantifier<K extends string, A, Rec extends ParentRec
     partial: FluentResult | undefined = undefined,
     depth = 0): FluentResult {
 
-    const example = partial || new FluentResult(!this.breakValue)
-
-    const collection = depth === 0 ?
-      this.cache :
-      (partial !== undefined ?
-        this.a.shrink(partial.example[this.name]).sampleUniqueWithBias(this.configuration.shrinkSize) :
-        [])
+    this.strategy.configArbitrary(this.name, partial, depth)
 
-    for (const tp of collection) {
-      testCase[this.name] = tp
+    while (this.strategy.hasInput(this.name)) {
```

Comment:
Could this be designed has an iterator so that we can use a `for ... of```?

Reply:
Yes, it could. In fact, it is basically what is happening behind the curtains. Do you have a reason for preferring an explicit `for`?

Reply:
It's already an iterator, why not make it explicit?

Reply:
I don't know, it just seemed more clearer to me this way.

Reply:
You can still use it this way, but being an iterator allows to be used with a `for ... of`.

Reply:
I agree but for future strategies where you generate one input at a time maybe it makes more sense the way I implemented I guess.